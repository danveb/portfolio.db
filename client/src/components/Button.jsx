import { Link } from "react-router-dom";
import rightArrow from "../assets/button/right-arrow.png"; 
import rightArrowAlt from "../assets/button/right-arrow-alt.png"; 
import "../styles/Button.css";

export default function Button({ title, url, rel=null, target=null }) {
  return (
    <div className="button__container">
      <Link to={url} className={url.indexOf("https://github.com") !== -1 ? "button--cta--alt" : "button--cta"} rel={rel} target={target}>
        {title}
        <img src={url.indexOf("https://github.com") !== -1 ? rightArrowAlt : rightArrow} alt="right arrow"/>
      </Link>
    </div>
  ) 
}