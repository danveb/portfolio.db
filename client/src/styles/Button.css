.button__container {
  width: 100%; 
  height: 100%; 
  max-width: fit-content; 
}

.button--cta, 
.button--cta--alt {
  font-size: 13px; 
  line-height: 1; 
  padding: 0.5rem; 
  display: flex; 
  flex-direction: row; 
  align-items: center; 
  gap: 15px; 
  text-decoration: none; 
}

.button--cta {
  color: #c7c9cb; 
  border: 1px solid var(--alice-blue); 
  background-color: transparent; 
}

.button--cta--alt {
  background-color: var(--alice-blue); 
  color: var(--raisin-black); 
  border: 1px solid var(--rich-black); 
}

.button--cta img, 
.button--cta--alt img {
  width: 15px; 
  height: 20px; 
  object-fit: contain; 
}
