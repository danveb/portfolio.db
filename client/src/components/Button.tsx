import { Link } from "react-router-dom";
import rightArrow from "../assets/button/right-arrow.png"; 
import rightArrowAlt from "../assets/button/right-arrow-alt.png"; 
import "../styles/Button.css";
import { ButtonProps } from "../types";

export default function Button({ title, url, rel, target }: ButtonProps) {
  return (
    <div className="button__container" data-testid="button">
      <Link to={url} className={url.indexOf("https://github.com") !== -1 ? "button--cta--alt" : "button--cta"} rel={rel} target={target}>
        {title}
        <img src={url.indexOf("https://github.com") !== -1 ? rightArrowAlt : rightArrow} alt="right arrow"/>
      </Link>
    </div>
  ) 
}