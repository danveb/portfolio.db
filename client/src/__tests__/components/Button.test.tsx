import { render, screen } from "@testing-library/react"; 
import Button from "../../components/Button"; 
import { BrowserRouter } from "react-router-dom";

// interface 
interface ButtonMockProps {
  title: string; 
  url: string; 
  alt: string; 
  rel?: (string); 
  target?: (string); 
}

// mocking...
// jest.mock("../../components/Button", () => {
//   return jest.fn(() => null) ; 
// }); 

// mocking v2 (hardcoding?) 
jest.mock("../../components/Button", () => (props: ButtonMockProps) => {
  return <button data-testid="button">
    <a href={props.url} >
      {props.title}
      <img src={props.url} alt={props.alt} />
    </a>
  </button>
})

describe("Button component", () => {
  // smoke test
  it("renders without crashing", () => {
    render(
      <BrowserRouter>
        <Button title="Hello World" url="https://google.com" />
      </BrowserRouter>
    ); 
  }); 

  // snapshot test
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Button title="Hello World" url="https://google.com" />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  });

  // getByTestId
  test("matches test id", () => {
    render(
      <BrowserRouter>
        <Button title="Hello World" url="https://google.com" />
      </BrowserRouter>
    );
    const testId = screen.getByTestId("button"); 
    expect(testId).toBeInTheDocument(); 
  });

  test("matches title tag", () => {
    render(
      <BrowserRouter>
        <Button title="Hello World" url="https://google.com" />
      </BrowserRouter>
    );
    const textTitle = screen.getByText("Hello World"); 
    expect(textTitle).toBeInTheDocument(); 
    expect(textTitle).not.toBe("World Hello"); 
  }); 
});