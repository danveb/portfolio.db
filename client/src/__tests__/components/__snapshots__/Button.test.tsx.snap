// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="button"
  >
    <a
      href="https://google.com"
    >
      Hello World
      <img
        src="https://google.com"
      />
    </a>
  </button>
</DocumentFragment>
`;
